src/AWS.Deploy.Orchestration/DeploymentBundleHandler.cs

```diff
@@ -108,21 +112,70 @@ public async Task PushDockerImageToECR(Recommendation recommendation, string rep
             recommendation.DeploymentBundle.ECRImageTag = tagSuffix;
         }
 
+        /// <summary>
+        /// The supported .NET versions on Elastic Beanstalk are dependent on the available platform versions.
+        /// These versions do not always have the required .NET runtimes installed so we need to perform extra checks
+        /// and perform a self-contained publish when creating the deployment bundle if needed.
+        /// </summary>
+        private void SwitchToSelfContainedBuildIfNeeded(Recommendation recommendation)
+        {
+            if (recommendation.Recipe.TargetService == RecipeIdentifier.TARGET_SERVICE_ELASTIC_BEANSTALK)
+            {
+                var targetFramework = recommendation.ProjectDefinition.TargetFramework ?? string.Empty;
```

Comment:
Could we return if `targetFramework` is null or empty? Not too concerned with the performance, but it'd help readability to not have think about how the missing value is handled through all of this?

Reply:
Also, are we skipping all of this if it's already set to self contained manually? If not, could we?

Reply:
Done